src/main/java/io/cryostat/net/OpenShiftAuthManager.java

```diff
@@ -287,6 +345,54 @@ private String getTokenFromHttpHeader(String rawHttpHeader) {
         }
     }
 
+    private Future<URI> computeAuthorizationEndpoint() {
```

Comment:
Is this cacheable? It doesn't appear to use any end user-specific information, and the inputs like environment variables won't change during the Cryostat process' lifetime.

Reply:
Yes this is cacheable. I tried using Guava at first but am still looking into using Caffeine instead to avoid adding another dependency.

Reply:
Yes definitely avoid adding a dependency just for this.

Since it's only a single value that needs to be cached/memoized you don't even need a whole Caffeine cache, since that would give a lot more power and flexibility with cache eviction policies etc. that aren't needed here. All we basically need is lazy initialization and memoization of the result. [`ConcurrentHashMap#computeIfAbsent`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#computeIfAbsent-K-java.util.function.Function-) is simple, adds no dependency, and probably suits all the requirements here well enough - just initialize it with a capacity of 1 and use some `static final` key for accessing the one memoized value.